modules/mono/mono_gd/gd_mono_internals.cpp

```diff
@@ -113,7 +113,7 @@ void tie_managed_to_unmanaged(MonoObject *managed, Object *unmanaged) {
 
 void unhandled_exception(MonoException *p_exc) {
 	mono_print_unhandled_exception((MonoObject *)p_exc);
-
+	mono_unhandled_exception((MonoObject *)p_exc);
 	if (GDMono::get_singleton()->get_unhandled_exception_policy() == GDMono::POLICY_TERMINATE_APP) {
```

Comment:
The call should go inside this first `if` block. In the the second one we simply log it and don't want any termination code.

Reply:
That makes sense since there seems to be no way to affect `UnhandledExceptionEventArgs.IsTerminating` property. But then there would be no way to have a hook on unhandled exception when we want to do something and don't want an application terminated. As far a I understand all that call really does is fires an event on default AppDomain.
Would it make more sense to add some global hook (i.e. event `GD.UnhandledException`) and call that in both cases while moving Mono's UnhandledException event to the first block?

Reply:
That sounds good to me, yes.